server/datastore/s3/installer.go

```diff
@@ -0,0 +1,62 @@
+package s3
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+)
+
+// InstallerParams describes the params used to locate an installer in an
+// S3 bucket
+type InstallerParams struct {
+	secret  string
+	ext     string
+	desktop bool
+}
+
+func (p InstallerParams) buildKey() string {
+	var dir string
+	if p.desktop {
+		dir = "desktop"
+	}
+	return path.Join(p.secret, dir, "fleet-osquery."+p.ext)
+}
+
+// InstallerStore contains methods to retrieve installers from S3
+type InstallerStore struct {
+	*datastore
+}
+
+// NewInstallerStore creates a new instance with the given S3 config
+func NewInstallerStore(config config.S3Config) (*InstallerStore, error) {
+	s3Store, err := newDatastore(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &InstallerStore{s3Store}, nil
+}
+
+// Exists checks if an installer exists in the S3 bucket
+func (s *InstallerStore) Exists(ctx context.Context, params InstallerParams) bool {
```

Comment:
I like that, I think I was just being lazy. For now I added

```go
if err != nil {
  return false, err
}
return true, nil
```

but we can get more descriptive with the return value, from the docs:

> A HEAD request has the same options as a GET action on an object. The response is identical to the GET response except that there is no response body. Because of this, if the HEAD request generates an error, it returns a generic 400 Bad Request, 403 Forbidden or 404 Not Found code. It is not possible to retrieve the exact exception beyond these error codes.